### PR TITLE
release: v1.57.0

### PR DESCRIPTION
**Pull request by abernix**
> **Note**
> **This particular PR must be true-merged to `main`.**

* This PR is only ready to review when it is marked as "Ready for Review".  It represents the merge to the `main` branch of an upcoming release (version number in the title).
* It will act as a staging branch until we are ready to finalize the release.
* We may cut any number of alpha and release candidate (RC) versions off this branch prior to formalizing it.
* This PR is **primarily a merge commit**, so reviewing every individual commit shown below is **not necessary** since those have been reviewed in their own PR.  However, things important to review on this PR **once it's marked "Ready for Review"**:
    - Does this PR target the right branch? (usually, `main`)
    - Are the appropriate **version bumps** and **release note edits** in the end of the commit list (or within the last few commits).  In other words, "Did the 'release prep' PR actually land on this branch?"
    - If those things look good, this PR is good to merge!

### CODE REVIEW

File: .changesets/feat_simon_replatform_introspection.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+### Remove legacy schema introspection ([PR #6139](https://github.com/apollographql/router/pull/6139))
+
+Schema introspection now runs natively without involving JavaScript code. We have high confidence that the two implementations return matching responses based on differential testing: fuzzing arbitrary queries against a large schema, and testing a corpus of customer schemas against a comprehensive query.
+
+In Router YAML configuration:
+
+* The `experimental_introspection_mode` key is removed, `new` is now the only behavior
+* The `supergraph.query_planning.legacy_introspection_caching` key is removed, the behavior is now similar to what was `false`: introspection responses are not part of the query plan cache but instead in a separate, small, in-memory only cache.
```
Comment (shorgi) [inline .changesets/feat_simon_replatform_introspection.md:8]:
```suggestion
Changes to the router's YAML configuration:

* The `experimental_introspection_mode` key has been removed, with the `new` mode as the only behavior in this release.
* The `supergraph.query_planning.legacy_introspection_caching` key is removed, with the behavior in this release now similar to what was `false`: introspection responses are not part of the query plan cache but instead in a separate, small in-memory—only cache.
```

File: .changesets/feat_simon_replatform_introspection.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+Migrations ensure that existing configuration files will keep working.
```
Comment (shorgi) [inline .changesets/feat_simon_replatform_introspection.md:10]: @SimonSapin does this mean the router supports existing config files with these removed keys automatically? Or do users need to do manual migration steps of removing/updating them?

File: .changesets/fix_feature_rhairouterrequesturi.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+### Add uri and method properties on router request in Rhai ([PR #6114](https://github.com/apollographql/router/pull/6114))
+
+Previously, when trying to access `request.uri` and `request.method` on a Router Request in Rhai, Router would error saying the properties are undefined.
```
Comment (shorgi) [inline .changesets/fix_feature_rhairouterrequesturi.md:3]:
```suggestion
### Support URI and method properties for router request in Rhai ([PR #6114](https://github.com/apollographql/router/pull/6114))

The router now supports accessing `request.uri` and `request.method` properties from custom Rhai scripts.

Previously, when trying to access `request.uri` and `request.method` on a router request in Rhai, the router would return error messages stating the properties were undefined.

An example Rhai script using these properties:
```

File: .changesets/fix_fix_typename.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+### Fix various edge cases for `__typename` field ([PR #6009](https://github.com/apollographql/router/pull/6009))
+
+The router now correctly handles the `__typename` field used on operation root types, even when the subgraph's root type has a name that differs from the supergraph's root type.
+
+For example, in query like this:
```
Comment (shorgi) [inline .changesets/fix_fix_typename.md:5]:
```suggestion
For example, given a query like this:

```

File: .changesets/maint_feature_rhaitelemetry.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+### Added telemetry for Rhai usage ([PR #6027](https://github.com/apollographql/router/pull/6027))
+
+Added telemetry for Rhai usage
```
Comment (shorgi) [inline .changesets/maint_feature_rhaitelemetry.md:3]:
```suggestion
The router now supports telemetry about custom Rhai scripts it's running. 
```

File: .changesets/maint_feature_rhaitelemetry.md
Comment (shorgi) [inline .changesets/maint_feature_rhaitelemetry.md:1]: @andrewmcgivery what's an example of using `apollo.router.config.rhai`?

Comment (andrewmcgivery) [reply .changesets/maint_feature_rhaitelemetry.md:1]: That change is actually internal Apollo feature telemetry, not something a customer would leverage. :)

File: .changesets/feat_rhai_uri_port.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
+### Support to get/set URI port in Rhai ([Issue #5437](https://github.com/apollographql/router/issues/5437))
```
[the remark this answers is not in the document]
Comment (abernix) [reply .changesets/feat_rhai_uri_port.md:1]: This still reads a bit funny to me — what about:

```suggestion
### Support reading `port` value of request URIs using Rhai ([Issue #5437](https://github.com/apollographql/router/issues/5437))
```

?

File: .changesets/maint_feature_rhaitelemetry.md
Comment (shorgi) [reply .changesets/maint_feature_rhaitelemetry.md:1]: thanks, then let's remove this from the external changelog

Comment (abernix) [reply .changesets/maint_feature_rhaitelemetry.md:1]: Agree.  done with https://github.com/apollographql/router/commit/d7c2e55f51be2ba994db0d61e1f00729e8ede05d

File: .changesets/feat_simon_replatform_introspection.md [hunk 7]
```diff
@@ -0,0 +1,12 @@
+### Remove legacy schema introspection ([PR #6139](https://github.com/apollographql/router/pull/6139))
+
+Schema introspection in the router now runs natively without JavaScript. We have high confidence that the new native implementation returns responses that match the previous Javascript implementation, based on differential testing: fuzzing arbitrary queries against a large schema, and testing a corpus of customer schemas against a comprehensive query.
+
+Changes to the router's YAML configuration:
+
+* The `experimental_introspection_mode` key has been removed, with the `new` mode as the only behavior in this release.
+* The `supergraph.query_planning.legacy_introspection_caching` key is removed, with the behavior in this release now similar to what was `false`: introspection responses are not part of the query plan cache but instead in a separate, small in-memory—only cache.
+
+Migrations ensure that existing configuration files will keep working.
```
Comment (abernix) [inline .changesets/feat_simon_replatform_introspection.md:10]:
```suggestion
If using these configuration options, automatic configuration migration will ensure that existing configuration works with no breaking changes, though updates would be necessary in the next major version of the router.  To simplify major upgrades, we recommend reviewing incremental updates to your YAML configuration by comparing the output of `./router config upgrade --config path/to/config.yaml` to your existing configuration.
```

**Review by BrynCooke — APPROVED** [review]

**Pull request merged**